client/rpc/src/state/mod.rs

```diff
@@ -598,7 +598,7 @@ where
 		})?;
 
 		// Returns proof of storage for child key entries at a specific block's state.
-		module.register_async_method("childstate_getChildReadProof", |params, state| {
+		module.register_async_method("state_getChildReadProof", |params, state| {
```

Comment:
Hmm, so this is weird. Are our aliases broken? To the outside they _should_ be the same right? FWIW I checked with the original author of this RPC and they said this was actually a mistake, it should have been `childstate_`.

If it has to be this way for some reason, then we should re-instate the alias to `childstate_`.

Reply:
it's didn't work because it was hardcoded to `state_getChildReadProof` in polkadot-js

Basically, if you add the alias `childstate_getChildReadProof` you get the method not decorated errors in polkadot-js, lemme double check

Reply:
Added an inline comment for this.

Reply:
How do you mean 'hardcoded'? I'd expect all method names to appear as strings in polkadot.js, or what am I missing?

On master today the method is named `state_getChildReadProof` and the alias is `childstate_getChildReadProof`. All I did was flipping them around. I would have thought that polkadot.js would not be able to distinguish between an alias and a "real" method name, so even if it's calling `state_getChildReadProof` (and hit the alias) it'd work anyways.

Is there a difference in how aliases are seen by external code? Is yes, isn't that a bug?